src/components/skills-quiz/SearchJobCard.jsx

```diff
@@ -1,78 +1,92 @@
 import PropTypes from 'prop-types';
 import Truncate from 'react-truncate';
 import Skeleton from 'react-loading-skeleton';
-// TODO: Uncomment this line when jobs are coming as hits from Algolia
-// import { camelCaseObject } from '@edx/frontend-platform/utils';
 import { Card } from '@edx/paragon';
 import { SearchContext } from '@edx/frontend-enterprise-catalog-search';
 
-const SearchJobCard = ({ hit, isLoading }) => { // eslint-disable-line no-unused-vars
+const getJobsToFetch = (jobs) => {
+  const jobsArray = [];
+  if (jobs) {
+    jobs.forEach(job => jobsArray.push(`name:${job}`));
+  }
+  return jobsArray;
+};
+
+const SearchJobCard = ({ index }) => {
   const { refinements } = useContext(SearchContext);
-  const { skill_names: skills } = refinements;
+  const { name: jobs } = refinements;
+  const [isLoading, setIsLoading] = useState(true);
+  const [selectedJobs, setSelectedJobs] = useState([]);
+  const jobsToFetch = getJobsToFetch(jobs);
 
-  // This statement will be usable once we have jobs data available in Algolia
-  // Currently we are showing skills data in place of that, 'hit' will be passed as props to SearchJobCard
-  // const job = hit ? camelCaseObject(hit) : {};
+  useEffect(
+    () => {
+      let fetch = true;
+      fetchJobs(); // eslint-disable-line no-use-before-define
+      return () => { fetch = false; };
+
+      async function fetchJobs() {
+        setIsLoading(true);
+        const { hits } = await index.search('', {
+          facetFilters: [
+            jobsToFetch,
+          ],
+        });
+        if (!fetch) { return; }
+        setSelectedJobs(hits);
+        setIsLoading(false);
+      }
+    },
+    [jobs],
+  );
 
   return (
-    <>
-      {skills?.map(skill => (
+    <div>
+      {selectedJobs?.map(job => (
         <div
-          key={skill}
+          key={job.name}
           className="search-job-card mb-3"
           role="group"
-          aria-label={skill}
+          aria-label={job.name}
         >
           <Card>
             <Card.Body>
-              <Card.Title as="h4" className="card-title mb-1">
+              <Card.Title as="h4" className="card-title mb-3">
                 {isLoading ? (
                   <Skeleton count={1} data-testid="job-title-loading" />
                 ) : (
                   <Truncate lines={1} trimWhitespace>
-                    {skill}
+                    {job.name}
                   </Truncate>
                 )}
               </Card.Title>
               {isLoading ? (
                 <Skeleton duration={0} data-testid="job-content-loading" />
               ) : (
                 <>
-                  <p className="text-muted m-0">
-                    <Truncate lines={1} trimWhitespace>
-                      {skill}
-                    </Truncate>
+                  <p className="text-muted m-0 medium-font">
+                    <span style={{ fontWeight: 700 }}>Median Salary:</span> {job.job_postings[0].median_salary}
```

Comment:
I think we should handle the case where a job has no job posting.

Reply:
Done